File: baekjoon/game/9655/Main.java
import java.util.*;

class Main {
    static int cache[];
    static int N;
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        cache = new int[1001];
        Arrays.fill(cache, -1);
        N = sc.nextInt();
        String ret = (play(N)==0?"SK":"CY");
        System.out.println(ret);
    }
    
    public static int play(int n) {
        if(cache[n] != -1) return cache[n];
        cache[n] = 0;
        if(n-3>0) {
            if(play(n-3) == 0) {
                cache[n] = 1;
            }
        } else if(n-1>0) {
            if(play(n-1) == 0) {
                cache[n] = 1;
            }
        }
        return cache[n];
    }
}

File: baekjoon/game/9655/main.cpp
#include <iostream>
#include <cstdio>
#include <cstring>

using namespace std;

int cache[1001];
int N;
int play(int n);
int main()
{
    memset(cache, -1, sizeof(cache));
    scanf("%d", &N);
    if(play(N) == 0) {
        printf("SK\n");
    } else {
        printf("CY\n");
    }
    return 0;
}

int play(int n)
{
    int& ret = cache[n];
    if(ret != -1) return ret;
    ret = 0;
    if(n-3 > 0) {
        if(!play(n-3)) {
            ret = 1;
        }
    } else if(n-1 > 0) {
        if(!play(n-1)) {
            ret = 1;
        }
    }
    return ret;
}